owr/owr.c

```diff
@@ -272,9 +272,21 @@ void owr_init_with_main_context(GMainContext *main_context)
     owr_init();
 }
 
+#if defined(__ANDROID__)
+static gboolean keep_alive_function(gpointer data) {
+    /* I keep the mainloop thread warm */
+    OWR_UNUSED(data);
+    return TRUE;
+}
+#endif
 
 static gpointer owr_run(gpointer data)
 {
+#if defined(__ANDROID__)
+    GSource *keep_alive_timer = g_timeout_source_new(1000);
+    g_source_set_callback(keep_alive_timer, keep_alive_function, NULL, NULL);
+    g_source_attach(keep_alive_timer, owr_main_context);
```

Comment:
Please report this against GLib and add a comment in the code here about *why* this is done, and link to the GLib bug report.

Ideally also try to reproduce it in a simple testcase. This is something I never saw with GLib on Android before and I was using GLib main contexts/loops there in many different situations.